src/raftstore/store/store.rs

```diff
@@ -430,6 +437,12 @@ impl<T: Transport, C: PdClient> Store<T, C> {
                     }
                 }
 
+                if is_leader {
```

Comment:
no matter, even the new_peer is not leader, the report region still has a larger version than pd's. Even another peer becomes leader, 1), it will still report the same region, 2), it will report a newer version region which can overwrite pd's. So it is ok here. 
Btw, the new_peer here will do campaign immediately, other peers will wait election timeout and do it, so most of time, the new_peer will become leader.